redash/tasks.py

```diff
@@ -240,10 +241,13 @@ def check_alerts_for_query(self, query_id):
     for alert in query.alerts:
         alert.query = query
         new_state = alert.evaluate()
-        if new_state != alert.state:
+        passed_rearm_threshold = False
+        if alert.rearm and alert.last_triggered_at:
+            passed_rearm_threshold = alert.last_triggered_at + datetime.timedelta(seconds=alert.rearm) < datetime.datetime.now().replace(tzinfo=alert.last_triggered_at.tzinfo)
+        if new_state != alert.state or (alert.state == models.Alert.TRIGGERED_STATE and passed_rearm_threshold ):
             logger.info("Alert %d new state: %s", alert.id, new_state)
             old_state = alert.state
-            alert.update_instance(state=new_state)
+            alert.update_instance(state=new_state, last_triggered_at=datetime.datetime.now())
```

Comment:
I think that if you use the `utcnow` function from `redash.utils` module here and on line 246, you won't need to replace the timezone.